### PR TITLE
Qlipp pipeline

### PR DESCRIPTION
**Pull request by camFoltz**
This PR makes some improvements and restructuring in the compute functions by vectorizing the reconstruction and getting rid of for loops.  I also restructured how the data is written and updated tests.  This fixed some bugs in 2D reconstruction

### CODE REVIEW

File: recOrder/compute/qlipp_compute.py [hunk 1]
```diff
@@ -155,28 +151,20 @@ def reconstruct_qlipp_birefringence(stokes, recon):
...
-        slices = 1
-        recon_data = np.zeros([1, 4, stokes.shape[-2], stokes.shape[-1]])
-
-    for z in range(slices):
-        recon_data[z, :, :, :] = recon.Polarization_recon(stokes[z] if slices != 1 else stokes)
-
+    birefringence = recon.Polarization_recon(stokes)
 
-    return np.transpose(recon_data, (1,0,2,3))
+    return np.transpose(birefringence, (0, 3, 1, 2)) if len(birefringence.shape) == 4 else birefringence
```
Comment (bryantChhun) [inline recOrder/compute/qlipp_compute.py:167]: Just for documentation -- I understand that a lot of the changes I see here are due to waveorder reconstruction needing dimensions like (C, Y, X, Z).

File: recOrder/pipelines/qlipp_pipeline.py [hunk 2]
```diff
@@ -54,10 +54,11 @@ def __init__(self, config: ConfigReader, data: MicromanagerReader, save_dir: str
         self.calib_meta = json.load(open(self.config.calibration_metadata)) \
             if self.config.calibration_metadata else None
         #todo: re-structure reading calib metadata when finalized
-        self.calib_scheme = self.calib_meta['Summary']['~ Acquired Using'] if self.calib_meta \
-            else '4-Frame Extinction'
+        self.calib_scheme = self.calib_meta['Summary']['Acquired Using'] if self.calib_meta \
+            else '4-Frame'
```
Comment (bryantChhun) [inline recOrder/pipelines/qlipp_pipeline.py:58]: I'm getting all `pre_post_processing` and `qlipp_pipeline` tests to fail at this line with:
```
E       KeyError: 'Acquired Using'

../recOrder/pipelines/qlipp_pipeline.py:57: KeyError
```

**Review by bryantChhun — CHANGES_REQUESTED** [review]
Tests fail when config file is parsed using "Acquired Using" as key.
Otherwise it looks good

File: recOrder/pipelines/qlipp_pipeline.py (hunk 2 above)
Comment (camFoltz) [reply recOrder/pipelines/qlipp_pipeline.py:58]: Yeah unfortunately the test dataset has a typo in the calibration_metadata file, I am going to create a new test dataset that has that fixed

Comment (camFoltz) [reply recOrder/pipelines/qlipp_pipeline.py]: Fixed.

**Review by bryantChhun — CHANGES_REQUESTED** [review]
one comment on docstring at write.

File: recOrder/pipelines/qlipp_pipeline.py [hunk 3]
```diff
@@ -179,29 +177,30 @@ def reconstruct_birefringence_volume(self, stokes):
...
         else:
-            birefringence = reconstruct_qlipp_birefringence(stokes[slice(None) if self.slices != 1 else self.focus_slice],
-                                                            self.reconstructor)
-            return birefringence
+            return reconstruct_qlipp_birefringence(
+                                stokes[:, :, :, slice(None) if self.slices != 1 else self.focus_slice],
+                                self.reconstructor)
 
     # todo: think about better way to write fluor/registered data?
     def write_data(self, pt, pt_data, stokes, birefringence, phase2D, phase3D, registered_stacks):
```
[the remark this answers is not in the document]
Comment (camFoltz) [reply recOrder/pipelines/qlipp_pipeline.py:218]: Done

[the remark this answers is not in the document]
Comment (camFoltz) [reply recOrder/pipelines/qlipp_pipeline.py:218]: The answer is no, and the pipeline interface does in fact require these values.

**Review by bryantChhun — APPROVED** [review]

**Pull request merged**